sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -217,10 +218,17 @@ private object PostgresDialect extends JdbcDialect with SQLConfHelper {
           case "42P07" =>
             // The message is: Failed to create index indexName in tableName
             val regex = "(?s)Failed to create index (.*) in (.*)".r
-            val indexName = regex.findFirstMatchIn(message).get.group(1)
-            val tableName = regex.findFirstMatchIn(message).get.group(2)
-            throw new IndexAlreadyExistsException(
-              indexName = indexName, tableName = tableName, cause = Some(e))
+            val maybeMatchIndex = regex.findFirstMatchIn(message)
```

Comment:
Nit: you can use regexes in match expressions, it'd be a little cleaner. See end of https://docs.scala-lang.org/tour/regular-expression-patterns.html

Reply:
thanks, LGTM